src/peerbook/libp2p_peerbook.erl

```diff
@@ -502,22 +502,27 @@ mk_this_peer(CurrentPeer, State=#state{tid=TID}) ->
 
 -spec update_this_peer(#state{}) -> #state{}.
 update_this_peer(State=#state{tid=TID}) ->
-    SwarmAddr = libp2p_swarm:pubkey_bin(TID),
-    case unsafe_fetch_peer(SwarmAddr, State#state.peerbook) of
-        {error, not_found} ->
-            NewPeer = mk_this_peer(undefined, State),
-            update_this_peer(NewPeer, get_async_signed_metadata(State));
-        {ok, OldPeer} ->
-            case mk_this_peer(OldPeer, State) of
-                {ok, NewPeer} ->
-                    case libp2p_peer:is_similar(NewPeer, OldPeer) of
-                        true -> State;
-                        false -> update_this_peer({ok, NewPeer}, get_async_signed_metadata(State))
-                    end;
-                {error, Error} ->
-                    lager:notice("Failed to make peer: ~p", [Error]),
-                    State
-            end
```

Comment:
Miner env variables in here seems like some serious name leakage? Can we change that?

Reply:
yeah, this was just something quick, I'll change it.